### PR TITLE
Add permissions to badges service

### PR DESCRIPTION
**Pull request by Wardormeur**

### CODE REVIEW

File: lib/perm/own-badge.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+function ownBadge (args, callback) {
+    var seneca = this;
+    var requestUser = args.user.id;
+    var badgeUser = args.params.badgeData.userId;
+    var validity = false;
```
Comment (DanielBrierton) [inline lib/perm/own-badge.js]: Why not just?
```
var validity = (requestUser === badgeUser);
```
or even forget about the validity var altogether and just go straight to
```
return callback(null, {'allowed': (requestUser === badgeUser)});
```

File: config/permissions.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+module.exports = function () {
+  return {
+    'cd-badges': {
+      'listBadges': [{
+        role: 'none'
+      }],
+
+      //  NOTE: Must be defined by visibility ?
```
Comment (DanielBrierton) [inline config/permissions.js]: Not quite sure what you mean by this comment

File: lib/perm/own-badge.js (hunk 1 above)
Comment (Wardormeur) [inline lib/perm/own-badge.js]: If you ensure me that equality cannot return anything than a boolean in Js, i sign :)

File: config/permissions.js (hunk 2 above)
Comment (Wardormeur) [inline config/permissions.js]: There is a couple of parameters on profile that defines visibility of fields on the public profile, that's somehting we should take into account maybe with a cp-visiblity ?

**Pull request merged**